karapace/schema_registry_apis.py

```diff
@@ -114,29 +128,30 @@ def _create_master_coordinator(self):
         self.mc = MasterCoordinator(config=self.config)
         self.mc.start()
 
-    def _subject_get(self, subject, content_type):
+    def _subject_get(self, subject, content_type, include_deleted=False):
         subject_data = self.ksr.subjects.get(subject)
         if not subject_data:
             self.r(
                 body={
                     "error_code": SchemaErrorCodes.SUBJECT_NOT_FOUND.value,
-                    "message": "Subject not found.",
+                    "message": f"Subject '{subject}' not found.",
                 },
                 content_type=content_type,
                 status=HTTPStatus.NOT_FOUND,
             )
 
-        schemas = self.ksr.get_schemas(subject)
+        schemas = self.ksr.get_schemas(subject, include_deleted=include_deleted)
         if not schemas:
             self.r(
                 body={
                     "error_code": SchemaErrorCodes.SUBJECT_NOT_FOUND.value,
-                    "message": "Subject not found.",
+                    "message": f"Subject '{subject}' not found.",
                 },
                 content_type=content_type,
                 status=HTTPStatus.NOT_FOUND,
             )
 
+        subject_data = subject_data.copy()
```

Comment:
Why is this now needed?

Reply:
It is in function `_subject_get` which without this modified internal data structures by eliminating soft deleted schema versions from memory permanently.  Ie. this is the one causing memory impact mentioned in the description.

Reply:
I see, so if called `include_deleted=true` this method would up pushing the deleted schemas to `ksr.subjects`? It was strange behaviour that this method has a mutating side effect. Sure that nothing depends on that? Now this method works differently, i.e. the side effect has been removed.

Reply:
If called with `include_deleted=` any value, it no longer mutates. Now it this method is free from side effects. Earlier any get method caused soft deleted schema versions in `ksr.subjects` to be removed, ie. `ksr.subjects` was modified both outside `KafkaSchemaReader` (`ksr`) and inside it.

I think the test suite should capture if this removal of side effect had any extra dependency. Of course beside the memory consumption mentioned.